pkg/server/share_manager.go

```diff
@@ -210,6 +240,96 @@ func (m *ShareManager) resizeVolume(devicePath, mountPath string) error {
 	return nil
 }
 
+func (m *ShareManager) makeLeaseClient() error {
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		m.logger.WithError(err).Warn("Failed to create InClusterConfig")
+		return err
+	}
+
+	clientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		m.logger.WithError(err).Warn("Failed to create kubernetes client set")
+		return err
+	}
+
+	// Use the clientset to get the node name of the share-manager pod
+	// and store for use by takeLease().
+	podName := shareManagerPrefix + m.volume.Name
+	pod, err := clientset.CoreV1().Pods(namespace).Get(m.context, podName, metav1.GetOptions{})
+	if err == nil && pod != nil {
+		m.leaseHolder = pod.Spec.NodeName
+	} else {
+		m.leaseHolder = podName
+	}
```

Comment:
So `leaseHolder` can be a node name OR a pod name?

Reply:
If a share-manager pod is not running, do we need to create lease holder?

Reply:
Agreed.  And share-manager pod is us, so there's a pretty good chance it is up.  The `else` path can go.